core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -3304,6 +3305,29 @@ public static RelNode createProject(RelNode child, Mappings.TargetMapping mappin
     return createProject(projectFactory, child, Mappings.asListNonNull(mapping.inverse()));
   }
 
+  /** Returns the relational table node for {@code tableName} if it occurs within a
+   * relational expression {@code root} otherwise an empty option is returned. */
+  public static Optional<RelOptTable> findTable(RelNode root, final String tableName) {
```

Comment:
I don't think we use `Optional` much in the project. I don't have a strong opinion for using it here or not but it seems that all the other methods in this class are simply returning null so for keeping things more consistent I would not add the wrapper.